heudiconv/heuristics/reproin.py

```diff
@@ -402,15 +402,43 @@ def infotodict(
     run_label: Optional[str] = None  # run-
     dcm_image_iod_spec: Optional[str] = None
     skip_derived = False
-    for curr_seqinfo in seqinfo:
+    for i_acq, curr_seqinfo in enumerate(seqinfo):
         # XXX: skip derived sequences, we don't store them to avoid polluting
         # the directory, unless it is the motion corrected ones
         # (will get _rec-moco suffix)
-        if skip_derived and curr_seqinfo.is_derived and not curr_seqinfo.is_motion_corrected:
+        if (
+            skip_derived
+            and curr_seqinfo.is_derived
+            and not curr_seqinfo.is_motion_corrected
+        ):
             skipped.append(curr_seqinfo.series_id)
             lgr.debug("Ignoring derived data %s", curr_seqinfo.series_id)
             continue
 
+        if i_acq == 0:
+            prev_dcm_image_iod_spec = None
+        else:
+            prev_seqinfo = seqinfo[i_acq - 1]
+            for f in "series_description", "protocol_name":
+                prev_seqinfo = prev_seqinfo._replace(
+                    **{f: getattr(prev_seqinfo, f).format(**prev_seqinfo._asdict())}
+                )
+
+            prev_dcm_image_iod_spec = prev_seqinfo.image_type[2]
+
+        if i_acq == (len(seqinfo) - 1):
+            next_series_description = None
+            next_dcm_image_iod_spec = None
+        else:
+            next_seqinfo = seqinfo[i_acq + 1]
+            for f in "series_description", "protocol_name":
+                next_seqinfo = next_seqinfo._replace(
+                    **{f: getattr(next_seqinfo, f).format(**next_seqinfo._asdict())}
+                )
+
+            next_series_description = next_seqinfo.series_description
+            next_dcm_image_iod_spec = next_seqinfo.image_type[2]
+
```

Comment:
this duplicates that `_replace` hack 2 more times, causes me duplication-allergy...

Since we are to do that anyways, do it once then -- just in a loop before this loop apply the replacements to all first.

Then avoid that `i_acq` - do following 'zip'ing, as chatgpt suggested

```
my_list = [1, 2, 3, 4, 5]

for previous, current, next in zip([None] + my_list[:-1], my_list, my_list[1:] + [None]):
    print(f"Previous: {previous}, Current: {current}, Next: {next}")
```

which would avoid all the if's and else's etc and make code easier to read IMHO.